### DISCUSSION TITLE
login is failing when setup proxy in next js app

### DISCUSSION THREAD

**Discussion by jp112233** [nextauthjs/next-auth#3183] [Help]
### Question 💬

I am using the next-auth library for authentication in my application. I am using fusionauth authentication using next-auth. 

- Everything is working fine while development and application was running over localhost:3000. Also, I have deployed my app in azure web app with a domain like abc.com and everything was working completely fine. 

- Now, In my case, I would need to serve my application over test.com. What we are doing is traverse requests from test.com --> abc.com. This has been set up using azure CDN(Also, I tried other approaches as well). 

- In this case, my login stopped working which was working fine in the first 2 scenarios. When I was debugging this issue, I came to know that It is failing at csrf token part. The token which is set in httponly cookie and token which is passed in API call of fusionauth login, both are different so next-auth is not validating this request and NOT redirecting user at login. On another note, this csrf token has been managed by next-auth itself. I am not using anything from end here! I just put loges in library itself and came to know these things!

Could you please look at this and provide your feedback for the same? Also please let me know If anything else is required from my end to debug this issue!

Thanks a lot in advance!!

### How to reproduce ☕️

1. Create a sample app.
2. implement login into the application.
3. Deploy it anywhere like azure or was or digital ocean VM with a domain like abc.com.
4. Setup a webserver or use CDN(in case of azure) to set up the application client URL and traverse request to abc.com. So basically your client should request to test.com and it redirects it to abc.com.
5. Login won't be working any more.

I will be adding a sample code repo URL soon.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by jp112233** [comment]
@balazsorban44 ,  Could you please help me with this issue?

Thanks a lot in advance.

**Comment by jp112233** [comment]
Can anyone please help me with this? 
That would be really great If you can suggest something here, please. @balazsorban44

**Comment by balazsorban44** [comment]
Maybe https://next-auth.js.org/configuration/options#nextauth_url_internal

**Comment by amaster507** [comment]
Your problem may be the same as mine but under a different approach. Your CDN may be caching incorrectly. I am not an Azure user, but see if there is a way to disable caching even if just temporarily to test this theory. If disabling caching resolves the issue, then look for a way to customize the config of this caching layer. Your CDN probably is not smart enough by default to know what data/pages from the underlying reverse proxy should be cached and is caching too much.
